src/storage/leveldb/history_index.ts

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import path from 'path';
+import { Level } from 'level';
+import { AbstractSublevel } from 'abstract-level';
+import { IKVHistoryIndex, IHistoryTx, HistoryIndexValidateResponse } from '../../types';
+import { errorCodeOrNull, KEY_NOT_FOUND_CODE } from './errors';
+
+export const HISTORY_PREFIX = 'history';
+export const TS_HISTORY_PREFIX = 'ts_history';
+
+
+function _ts_key(tx: Pick<IHistoryTx, 'timestamp'|'tx_id'>): string {
+  const buf = Buffer.alloc(4);
+  buf.writeUint32BE(tx.timestamp);
+  return `${buf.toString('hex')}:${tx.tx_id}`;
+}
+
+export default class LevelHistoryIndex implements IKVHistoryIndex {
+  dbpath: string;
+  historyDB: AbstractSublevel<Level, string | Buffer | Uint8Array, string, IHistoryTx>;
+  tsHistoryDB: AbstractSublevel<Level, string | Buffer | Uint8Array, string, IHistoryTx>;
+  size: number;
+  indexVersion: string = '0.0.1';
+
+  constructor(dbpath: string) {
+    this.dbpath = path.join(dbpath, 'history');
+    const db = new Level(this.dbpath);
+    this.historyDB = db.sublevel<string, IHistoryTx>(HISTORY_PREFIX, { valueEncoding: 'json' });
+    this.tsHistoryDB = db.sublevel<string, IHistoryTx>(TS_HISTORY_PREFIX, { valueEncoding: 'json' });
```

Comment:
It is not needed, this index is used as an iteration index.
With it we can iterate on all transactions ordered by the timestamp.

Reply:
How do you iterate by timestamp in the `MemoryStore`?

Reply:
Since we use `Map`s in `MemoryStore` to store the history instead of `Record`s the iterator guarantees iteration on order of inserts.
To guarantee timestamp ordering on memory store i was thinking of using an auxiliar [SortedMap](https://www.collectionsjs.com/sorted-map) but this would require a new dependency so I still need to check the implementation of this `SortedMap`.

Reply:
Have you created this issue so we can handle it as soon as it's merged?

Reply:
The issue #480 was created to track this discussion.